### DISCUSSION TITLE
0204 timestep 3 simulation issues

### DISCUSSION THREAD

**Discussion by HettiePike** [IBAHCM/RPiR#200] [Archive] [answered]
Hi @IBAHCM/rpir-instructors , 

I'm having issues proving that the two functions are the same for 0204 . My simulation overlaps for timestep<-0.5 but not at 3 and I have no idea what the difference is in my code to make that happened? 

Here's the link to the script: 
https://github.com/IBAHCM/hettiepike_series02.git

And this is what the plot currently looks like:
![0204 timestep3](https://user-images.githubusercontent.com/77628798/106695123-df11fb80-65d1-11eb-9937-717ea33409f0.png)

Thanks

**Comment by soniamitchell** [comment, marked as answer]
Hi @HettiePike, I've just run your code and I think I've found the problem. Try this:

1. Run lines 9-49. Now look at `timestep.herd.df`, it should look like this:

```
     time susceptible infected
1    0          98        2
```

Run lines 51-54 and look at `latest.population`, it looks like this:

```
    susceptible infected
1    97.35333 2.646667
```

After running line 55 you get the error:

```
Error in rbind(deparse.level, ...) : 
  numbers of columns of arguments do not match
```

Why? because `rbind()` is trying to stick `timestep.herd.df` to `latest.population`. 

The output of `timestep_deterministic_SIS` should match the format of `timestep.herd.df`, and so if you add a new column for `time` in its output (0204-step-deterministic-SIS.R, line 40), your problem should be fixed.

**Reply by HettiePike** [reply]
That's great thank you so much, @soniamitchell ! I have another issue though haha
For 0205 run, I keep getting this error, and I can't find where the fault is again. Would you possibly be able to have a look for me please?
https://github.com/IBAHCM/hettiepike_series02.git
Error in timestep_deterministic_SIR(latest = latest.populationA, transmission.rate = fmd.transmissionA,  : 
  object 'next.recovereds' not found
Thank you

**Reply by soniamitchell** [reply]
Hi @HettiePike, the error is telling you that R is looking for the object `next.recovereds` but it can't find it. So I've done a search of your code and found that you're using an object called `next.recovereds` in 0205-deterministic-SIS.R.

The problem is, you create `next.recovereds` on line 39, after having called it on line 38. You need to create an object first before using it. Does this make sense?

**Reply by richardreeve** [reply]
@soniamitchell is right, but I think in fact you are getting confused between next.recovereds and new.recovereds in line 38.

**Reply by HettiePike** [reply]
Ahh yep, got it, thanks!
@richardreeve , I see that now too. Thank you!
